src/build_tools/open.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+# Copyright 2010-2021, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""A cross-platform script to open file with resolving symlink.
+
+Usage:
+  % python3 build_tools/open.py bazel-bin/win32/installer/Mozc64.msi
+"""
+
+import argparse
+import os
+import pathlib
+import subprocess
+
+
+def is_windows() -> bool:
+  """Returns true if the platform is Windows."""
+  return os.name == 'nt'
+
+
+def is_mac() -> bool:
+  """Returns true if the platform is Mac."""
+  return os.name == 'posix' and os.uname()[0] == 'Darwin'
+
+
+def is_linux() -> bool:
+  """Returns true if the platform is Linux."""
+  return os.name == 'posix' and os.uname()[0] == 'Linux'
+
+
+def main():
+  parser = argparse.ArgumentParser()
+  parser.add_argument('filename', type=str)
+
+  args = parser.parse_args()
+
+  if args.filename == None:
+    print('Please specify the filename.')
+    return
+
+  filename = pathlib.Path(args.filename)
+  abspath = filename.resolve()
+
+  if not abspath.exists():
+    if abspath == filename:
+      print(f'The specified file {filename} does not exist.')
+    else:
+      print(f'The specified file {filename} ({abspath}) does not exist.')
+    return
+
+  command = []
+  if is_windows():
+    command = ['cmd', '/c', 'start', str(abspath)]
+  elif is_mac():
+    command = ['open', str(abspath)]
+  elif is_linux():
+    command = ['xdg-open', str(abspath)]
+  else:
+    print('Unsupported platform.')
+    return
+  subprocess.run(command, shell=False)
```

Comment:
Please set the `check` argument.

https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/subprocess-run-check.html

Reply:
Done.